### PR TITLE
feat: Add `copilot-chat-explain-symbol-at-line` function

### PR DESCRIPTION
**Pull request by tninja**
This function will explain the symbol under point. It can be used to check the public functions / variables developer not familiar with, so that developer can stay inside emacs and get the info, and don't have to google the symbol and read the document.

Let me know if you have any thought. Thanks.

### CODE REVIEW

File: copilot-chat.el [hunk 1]
```diff
@@ -423,8 +440,9 @@ Argument PROMPT is the prompt to send to Copilot."
                       nil
                       (if (= 0 copilot-chat--prompt-history-position)
                         ""
-                        (setq copilot-chat--prompt-history-position (1- copilot-chat--prompt-history-position))
-                        (nth copilot-chat--prompt-history-position copilot-chat--prompt-history))))))
+                        (progn
```
Comment (tninja) [inline copilot-chat.el:443]: the previous implementation is definitely right. I just add the progn for the else clause, it might improve the readability

**Comment by chep** [comment]
Thank you. While reviewing I found a bug with shell-maker and `(copilot-chat--prepare-buffers)`. I'm trying to fix it before merging this.

File: copilot-chat.el [hunk 2]
```diff
@@ -274,6 +274,26 @@ Argument PROMPT is the prompt to send to Copilot."
...
+         (lang (replace-regexp-in-string
+                "-mode$" ""
+                (symbol-name major-mode)))
+         (prompt (format "In %s programming language, please explain what '%s' means in the context of this code line:\n%s" 
+                        lang symbol line)))
+    (copilot-chat--prepare-buffers)
+    (with-current-buffer copilot-chat--prompt-buffer
+      (erase-buffer)
+      (insert prompt))
+    (copilot-chat-prompt-send)))
```
Comment (chep) [inline copilot-chat.el:296]: Could you move this to a `(copilot-chat--explain-symbol-at-line) function like I just did for `(copilot-chat-display)` ?
So I will be able to create an advice for shell-maker.
Thank you

Comment (chep) [reply copilot-chat.el:296]: And a function to create the prompt so I can also use it in shell-maker without any code redundancy.
Thank you.

Comment (tninja) [reply copilot-chat.el:296]: Thanks for rely. Do you mean this change? https://github.com/chep/copilot-chat.el/pull/35/files. There is advice-add and advice-remove, do you mean add those part?

Would be great to have more detail change suggestion, Thanks.

Comment (tninja) [reply copilot-chat.el:296]: Not sure if you mean this commit: https://github.com/chep/copilot-chat.el/pull/34/commits/2488c1eab5b74a54c90cb0b7721b574f5d167f1b

Comment (chep) [reply copilot-chat.el:296]: I just need the `(copilot-chat--explain-symbol-at-line)` function to be split and the prompt to be generated by a function :

``` emacs-lisp
(defun copilot-chat--explain-symbol-prompt()
  "Returns explain symbol prompt."
  (let ((lang (replace-regexp-in-string
               "-mode$" ""
               (symbol-name major-mode))))
    (format "In %s programming language, please explain what '%s' means in the context of this code line:\n%s" 
            lang symbol line)))

(defun copilot-chat--explain-symbol-at-line(prompt)
  "Ask Copilot to explain symbol under point.
Code line is given as background info.
Frontend may override this function.
PROMPT is the prompt to send to copilot"
    (copilot-chat--prepare-buffers)
    (with-current-buffer copilot-chat--prompt-buffer
      (erase-buffer)
      (insert prompt))
    (copilot-chat-prompt-send)))

;;;###autoload
(defun copilot-chat-explain-symbol-at-line()
  "Ask Copilot to explain symbol under point.
Code line is given as background info.
  (interactive)
  (unless (copilot-chat--ready-p)
    (copilot-chat-reset))
  (let* ((symbol (thing-at-point 'symbol))
         (line (buffer-substring-no-properties 
                (line-beginning-position)
                (line-end-position)))
        (prompt (copilot-chat--explain-symbol-prompt)))
  (copilot-chat-explain-symbol-at-line prompt))
  
```

So I can create an advice for shell maker frontend, something like this:
``` emacs-lisp
(defun copilot-chat--shell-maker-explain-symbol-at-line(prompt)
    (insert-in-shell-maker buffer prompt)
  (send-shell-maker-prompt))
  
```

The idea is to put everything not frontend dependent in the interactive function and everything else in a separate one.

(I wrote the above code without any test nor compilation)

Maybe the `(copilot-chat--explain-symbol-at-line)` function could just be named `(insert-and-send-prompt)` and could be reused in many other functions but I'll do that job ;-)

Comment (tninja) [reply copilot-chat.el]: I see. That make sense to separate prompt generation and send prompt to copilot-chat. Just did that and verified it still work. Also revert the previous change. Would you mind take a look again?

**Comment by chep** [comment]
Very nice, thank you a lot for your patience.

**Comment by chep** [comment]
I added the feature for shell-maker and code is now simpler. It will also be simpler to add new features.
Thanks again.

**Pull request merged**